activation/nipost_test.go

```diff
@@ -205,19 +206,21 @@ func buildNIPost(tb testing.TB, r *require.Assertions, postCfg PostConfig, nipos
 	poetProver, err := NewHTTPPoetHarness(true)
 	r.NoError(err)
 	r.NotNil(poetProver)
-	defer func() {
-		err = poetProver.Teardown(true)
-		r.NoError(err)
-	}()
+	tb.Cleanup(func() {
+		err := poetProver.Teardown(true)
+		if assert.NoError(tb, err, "failed to tear down harness") {
+			tb.Log("harness torn down")
+		}
+	})
 
 	var postProvider PostSetupProvider
 	postProvider, err = NewPostSetupManager(minerID, postCfg, logtest.New(tb))
 	r.NoError(err)
 	r.NotNil(postProvider)
-	defer func() {
+	tb.Cleanup(func() {
 		r.NoError(postProvider.LastError())
 		r.NoError(postProvider.StopSession(true))
-	}()
+	})
```

Comment:
Just being curious - why plain `defer` was not enough?

Reply:
- The time of execution and the guarantees are different: `tb.Cleanup` is executed at the end of the test, `defer` when the function returns.
- `tb.Cleanup` is also executed when a test panics, `defer` isn't (unless one `recover`s withn the `defer`). 
-  `t.Cleanup` gives an indication about the intend of the code

Here in this specific case it probably doesn't matter if one uses `t.Cleanup` or `defer`.